From stern@rowland.harvard.edu  Tue Aug 18 09:51:16 2009
From: Alan Stern <stern@rowland.harvard.edu>
Date: Mon, 3 Aug 2009 11:05:59 -0400 (EDT)
Subject: USB: s3c2410: unregister should call unbind, not disconnect
To: Greg KH <greg@kroah.com>
Message-ID: <Pine.LNX.4.44L0.0908031104590.16208-100000@iolanthe.rowland.org>


This patch (as1275) fixes the s3c2410 device controller driver.  Its
usb_gadget_unregister_driver() routine is supposed to call the gadget
driver's unbind method, not the disconnect method.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/s3c2410_udc.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/gadget/s3c2410_udc.c
+++ b/drivers/usb/gadget/s3c2410_udc.c
@@ -1703,8 +1703,7 @@ int usb_gadget_unregister_driver(struct 
 	dprintk(DEBUG_NORMAL,"usb_gadget_register_driver() '%s'\n",
 		driver->driver.name);
 
-	if (driver->disconnect)
-		driver->disconnect(&udc->gadget);
+	driver->unbind(&udc->gadget);
 
 	device_del(&udc->gadget.dev);
 	udc->driver = NULL;
